docs/guide/time.rst

```diff
@@ -16,29 +16,37 @@ SunPy builds upon its functionality.
 
 Solar data is associated with a number of different time formats. SunPy provides a simple
 parsing function which can deal with most every format that a user may encounter. Called
-`sunpy.time.parse_time()`, this function takes a string as input and returns a datetime object.
+`sunpy.time.parse_time()`, this function takes a string as input and returns a `~astropy.time.Time` object.
```

Comment:
This function takes much more than a string. Probably worth mentioning here. I also feel like this section generally undersells the utility of `parse_time` it dosen't really make mention of it supporting as many different and weird time formats as possible, I feel like it could do with a general spruce up.